flake.nix

```diff
@@ -11,12 +11,25 @@
           haskellNix.overlay
           (final: prev: {
             myHaskellProject =
-              final.haskell-nix.cabalProject' {
+              final.haskell-nix.cabalProject' ({
                 src = ./haskell;
                 compiler-nix-name = "ghc924";
                 shell.tools = { cabal = { }; };
                 shell = { inherit crossPlatforms; };
-              };
+              }
+               // (if pkgs.stdenv.hostPlatform.isMusl then
+                  {
+                    configureArgs =
+                      ''
+                        --disable-executable-dynamic
+                        --disable-shared
+                        --ghc-option=-optl=-pthread
+                        --ghc-option=-optl=-static
+                        --ghc-option=-optl=-L${pkgs.gmp6.override { withStatic = true; }}/lib
+                        --ghc-option=-optl=-L${pkgs.zlib.static}/lib
+                      '';
```

Comment:
This isn't actually tested, because it doesn't affect the dev shell. And as mentioned elsewhere, the Musl build has issues with `libevdev`.

Reply:
If we enable the `release` flag in this file, we need to be careful not to overwrite that here.

Reply:
Unnecessary, as we've now shown by having a working musl build.